opal/runtime/opal_params_core.c

```diff
@@ -89,10 +89,11 @@ static bool opal_register_util_done = false;
 
 static char *opal_var_dump_color_string = NULL;
 
-static char *opal_var_dump_color_keys[OPAL_VAR_DUMP_COLOR_KEY_COUNT] = {
+static char *opal_var_dump_color_keys[OPAL_VAR_DUMP_COLOR_KEY_COUNT + 1] = {
```

Comment:
The size should also be changed in the header file: https://github.com/open-mpi/ompi/blob/e3a0f7b1cb7054aa863b3a4d490cc66cf803fd74/opal/runtime/opal_params_core.h#L49

Reply:
Thanks! I was thinking of that, but forgot o check! Good catch

Reply:
@devreal should be fixed now

Reply:
@gkatev What is the relation between `opal_var_dump_color_keys` and `opal_var_dump_color`?  Is a larger fix needed here?

Reply:
Ahh shoot, those are two different arrays 🤦  Not sure `opal_var_dump_color` needs to change, thanks for pointing that out @jsquyres

Reply:
color values do not seem considered NULL-terminated, so maybe they don't need an extra element

Reply:
No, as far as I can tell only the keys array is affected. The keys/values arrays weren't meant to be NULL-terminated.

Unless I'm missing something, the only points that the two are passed to opal_argv functions are the two I mentioned in https://github.com/open-mpi/ompi/issues/11826#issuecomment-1639927165.

Now, if it proves too peculiar trying to shape them so that they are NULL terminated, I would suggest just replacing the two offending calls. I can put together a patch tomorrow and offer it for consideration.